exceptions/exceptions.hh

```diff
@@ -65,6 +65,8 @@ enum class exception_code : int32_t {
     WRITE_FAILURE   = 0x1500,
     READ_TIMEOUT    = 0x1200,
     READ_FAILURE    = 0x1300,
+    FUNCTION_FAILURE= 0x1400,
```

Comment:
Nit for a follow-up: please sort.

Reply:
6ef26a4a4a6de3b0b46bb0b08b083761f90653ee added `WRITE_FAILURE`  disordered so it be close to WRITE_TIMEOUT.
But given the additional values sorting all of them does make more sense.